# Dementia Detection in Alzheimer's Patients

## 1. Business Understanding

### 1.1 Overview

### 1.2 Problem Statement

### 1.3 Business Objectives

### 1.4 Research Questions

### 1.5 Success Criteria


## Data Understanding

The objectives for this 

Column Descriptions:

Subject.ID: Unique identifier for the Alzheimer's patients.

MRI.ID: Unique identifier for the MRI scans

Group (Converted / Demented / Nondemented): The column tells whether the Alzheimers patient is demented or not also whether the patient was the patient was non demented but was diagnosed as demented after a couple of visits.

Visit - Number of visit: The number of times the patient 

MR.Delay

Demographics Info

- M.F - Gender of the patient. 
- Hand - Handedness (actually all subjects were right-handed so I will drop this column)
- Age - Age of the patient
- EDUC - Years of education
- SES - Socioeconomic status as assessed by the Hollingshead Index of Social Position and classified into categories from 1 (highest status) to 5 (lowest status)

Clinical Info

- MMSE - Mini-Mental State Examination score (range is from 0 = worst to 30 = best)
- CDR - Clinical Dementia Rating (0 = no dementia, 0.5 = very mild dementia, 1 = mild dementia, 2 = moderate dementia)

Derived Anatomic Volumes

- eTIV - Estimated total intracranial volume, mm3
- nWBV - Normalized whole-brain volume, expressed as a percent of all voxels in the atlas-masked image that are labeled as gray or white matter by the automated tissue segmentation process
- ASF - Atlas scaling factor (unitless). Computed scaling factor that transforms native-space brain and skull to the atlas target (i.e., the determinant of the transform matrix)



#### Importing Necessary Packages

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression

In [2]:
dataset_1 = pd.read_csv('demo_data/oasis_cross-sectional.csv')
dataset_2 = pd.read_csv('demo_data/oasis_longitudinal.csv')

In [3]:

class DatasetChecker:
        def __init__(self, dataset):
            self.dataset = dataset
            
        def check_first_rows(self):
            # Print the first five rows of the dataset
            print("First five rows:")
            return self.dataset.head()
        
        def check_last_rows(self):
            print("Last five rows:")
            return self.dataset.tail()
        
        def check_shape(self):
            # Print the shape of the dataset
            print("Shape of dataset")
            return self.dataset.shape
        
        def check_info(self):
            # print information on dataset
            print("Information on dataset:")
            return self.dataset.info()
        
        def check_description(self):
            # print the summary statistics of dataset
            print("Summary statistics")
            return self.dataset.describe()
        
        def check_data_types(self):
            # Print the data types of the columns in the dataset
            print("Data types:")
            return self.dataset.dtypes
        
        def check_missing_values(self):
            # Print the number of missing values
            print("Number of missing values")
            return self.dataset.isna().sum()
        
            
# Create a list of methods from the DatasetChecker class

methods = [DatasetChecker.check_first_rows, DatasetChecker.check_last_rows, DatasetChecker.check_shape,
           DatasetChecker.check_info, DatasetChecker.check_description, 
           DatasetChecker.check_data_types,DatasetChecker.check_missing_values]


In [4]:
# Create an instance of the DatasetChecker class for the first dataset
checker1 = DatasetChecker(dataset_1)

# Use a for loop to iterate over the methods in the list
for method in methods:
  # Call the method on the instance, passing no arguments
    result = method(checker1)
    display(result)

First five rows:


ID M/F Hand  Age  Educ  SES  MMSE  CDR  eTIV   nWBV    ASF  \
0  OAS1_0001_MR1   F    R   74   2.0  3.0  29.0  0.0  1344  0.743  1.306   
1  OAS1_0002_MR1   F    R   55   4.0  1.0  29.0  0.0  1147  0.810  1.531   
2  OAS1_0003_MR1   F    R   73   4.0  3.0  27.0  0.5  1454  0.708  1.207   
3  OAS1_0004_MR1   M    R   28   NaN  NaN   NaN  NaN  1588  0.803  1.105   
4  OAS1_0005_MR1   M    R   18   NaN  NaN   NaN  NaN  1737  0.848  1.010   

   Delay  
0    NaN  
1    NaN  
2    NaN  
3    NaN  
4    NaN

Last five rows:


ID M/F Hand  Age  Educ  SES  MMSE  CDR  eTIV   nWBV    ASF  \
431  OAS1_0285_MR2   M    R   20   NaN  NaN   NaN  NaN  1469  0.847  1.195   
432  OAS1_0353_MR2   M    R   22   NaN  NaN   NaN  NaN  1684  0.790  1.042   
433  OAS1_0368_MR2   M    R   22   NaN  NaN   NaN  NaN  1580  0.856  1.111   
434  OAS1_0379_MR2   F    R   20   NaN  NaN   NaN  NaN  1262  0.861  1.390   
435  OAS1_0395_MR2   F    R   26   NaN  NaN   NaN  NaN  1283  0.834  1.368   

     Delay  
431    2.0  
432   40.0  
433   89.0  
434    2.0  
435   39.0

Shape of dataset


(436, 12)

Information on dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 436 entries, 0 to 435
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   ID      436 non-null    object 
 1   M/F     436 non-null    object 
 2   Hand    436 non-null    object 
 3   Age     436 non-null    int64  
 4   Educ    235 non-null    float64
 5   SES     216 non-null    float64
 6   MMSE    235 non-null    float64
 7   CDR     235 non-null    float64
 8   eTIV    436 non-null    int64  
 9   nWBV    436 non-null    float64
 10  ASF     436 non-null    float64
 11  Delay   20 non-null     float64
dtypes: float64(7), int64(2), object(3)
memory usage: 41.0+ KB


None

Summary statistics


Age        Educ         SES       MMSE         CDR         eTIV  \
count  436.000000  235.000000  216.000000  235.00000  235.000000   436.000000   
mean    51.357798    3.178723    2.490741   27.06383    0.285106  1481.919725   
std     25.269862    1.311510    1.120593    3.69687    0.383405   158.740866   
min     18.000000    1.000000    1.000000   14.00000    0.000000  1123.000000   
25%     23.000000    2.000000    2.000000   26.00000    0.000000  1367.750000   
50%     54.000000    3.000000    2.000000   29.00000    0.000000  1475.500000   
75%     74.000000    4.000000    3.000000   30.00000    0.500000  1579.250000   
max     96.000000    5.000000    5.000000   30.00000    2.000000  1992.000000   

             nWBV         ASF     Delay  
count  436.000000  436.000000  20.00000  
mean     0.791670    1.198894  20.55000  
std      0.059937    0.128682  23.86249  
min      0.644000    0.881000   1.00000  
25%      0.742750    1.111750   2.75000  
50%      0.809000    1.190000  11.00000  
75%      0.842000    1.284250  30.75000  
max      0.893000    1.563000  89.00000

Data types:


ID        object
M/F       object
Hand      object
Age        int64
Educ     float64
SES      float64
MMSE     float64
CDR      float64
eTIV       int64
nWBV     float64
ASF      float64
Delay    float64
dtype: object

Number of missing values


ID         0
M/F        0
Hand       0
Age        0
Educ     201
SES      220
MMSE     201
CDR      201
eTIV       0
nWBV       0
ASF        0
Delay    416
dtype: int64

#### Observations

- The dataset has 436 rows and 12 columns. 
- Both the head and tail have the same number of columns indicating that our dataset is consistent.
- Out of the 12 columns, 9 are numeric while 3 are of object type.
- The following columns have missing values:
    - Educ - 201
    - SES - 220
    - MMSE - 201
    - CDR - 201
    - Delay - 416

In [5]:
# Create an instance of the DatasetChecker class for the second dataset
checker2 = DatasetChecker(dataset_2)

# Use a for loop to iterate over the methods in the list
for method in methods:
  # Call the method on the instance, passing no arguments
    result = method(checker2)
    display(result)

First five rows:


Subject ID         MRI ID        Group  Visit  MR Delay M/F Hand  Age  EDUC  \
0  OAS2_0001  OAS2_0001_MR1  Nondemented      1         0   M    R   87    14   
1  OAS2_0001  OAS2_0001_MR2  Nondemented      2       457   M    R   88    14   
2  OAS2_0002  OAS2_0002_MR1     Demented      1         0   M    R   75    12   
3  OAS2_0002  OAS2_0002_MR2     Demented      2       560   M    R   76    12   
4  OAS2_0002  OAS2_0002_MR3     Demented      3      1895   M    R   80    12   

   SES  MMSE  CDR  eTIV   nWBV    ASF  
0  2.0  27.0  0.0  1987  0.696  0.883  
1  2.0  30.0  0.0  2004  0.681  0.876  
2  NaN  23.0  0.5  1678  0.736  1.046  
3  NaN  28.0  0.5  1738  0.713  1.010  
4  NaN  22.0  0.5  1698  0.701  1.034

Last five rows:


Subject ID         MRI ID        Group  Visit  MR Delay M/F Hand  Age  \
368  OAS2_0185  OAS2_0185_MR2     Demented      2       842   M    R   82   
369  OAS2_0185  OAS2_0185_MR3     Demented      3      2297   M    R   86   
370  OAS2_0186  OAS2_0186_MR1  Nondemented      1         0   F    R   61   
371  OAS2_0186  OAS2_0186_MR2  Nondemented      2       763   F    R   63   
372  OAS2_0186  OAS2_0186_MR3  Nondemented      3      1608   F    R   65   

     EDUC  SES  MMSE  CDR  eTIV   nWBV    ASF  
368    16  1.0  28.0  0.5  1693  0.694  1.037  
369    16  1.0  26.0  0.5  1688  0.675  1.040  
370    13  2.0  30.0  0.0  1319  0.801  1.331  
371    13  2.0  30.0  0.0  1327  0.796  1.323  
372    13  2.0  30.0  0.0  1333  0.801  1.317

Shape of dataset


(373, 15)

Information on dataset:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 373 entries, 0 to 372
Data columns (total 15 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Subject ID  373 non-null    object 
 1   MRI ID      373 non-null    object 
 2   Group       373 non-null    object 
 3   Visit       373 non-null    int64  
 4   MR Delay    373 non-null    int64  
 5   M/F         373 non-null    object 
 6   Hand        373 non-null    object 
 7   Age         373 non-null    int64  
 8   EDUC        373 non-null    int64  
 9   SES         354 non-null    float64
 10  MMSE        371 non-null    float64
 11  CDR         373 non-null    float64
 12  eTIV        373 non-null    int64  
 13  nWBV        373 non-null    float64
 14  ASF         373 non-null    float64
dtypes: float64(5), int64(5), object(5)
memory usage: 43.8+ KB


None

Summary statistics


Visit     MR Delay         Age        EDUC         SES  \
count  373.000000   373.000000  373.000000  373.000000  354.000000   
mean     1.882038   595.104558   77.013405   14.597855    2.460452   
std      0.922843   635.485118    7.640957    2.876339    1.134005   
min      1.000000     0.000000   60.000000    6.000000    1.000000   
25%      1.000000     0.000000   71.000000   12.000000    2.000000   
50%      2.000000   552.000000   77.000000   15.000000    2.000000   
75%      2.000000   873.000000   82.000000   16.000000    3.000000   
max      5.000000  2639.000000   98.000000   23.000000    5.000000   

             MMSE         CDR         eTIV        nWBV         ASF  
count  371.000000  373.000000   373.000000  373.000000  373.000000  
mean    27.342318    0.290885  1488.128686    0.729568    1.195461  
std      3.683244    0.374557   176.139286    0.037135    0.138092  
min      4.000000    0.000000  1106.000000    0.644000    0.876000  
25%     27.000000    0.000000  1357.000000    0.700000    1.099000  
50%     29.000000    0.000000  1470.000000    0.729000    1.194000  
75%     30.000000    0.500000  1597.000000    0.756000    1.293000  
max     30.000000    2.000000  2004.000000    0.837000    1.587000

Data types:


Subject ID     object
MRI ID         object
Group          object
Visit           int64
MR Delay        int64
M/F            object
Hand           object
Age             int64
EDUC            int64
SES           float64
MMSE          float64
CDR           float64
eTIV            int64
nWBV          float64
ASF           float64
dtype: object

Number of missing values


Subject ID     0
MRI ID         0
Group          0
Visit          0
MR Delay       0
M/F            0
Hand           0
Age            0
EDUC           0
SES           19
MMSE           2
CDR            0
eTIV           0
nWBV           0
ASF            0
dtype: int64

#### Observations

- The dataset has 373 rows and 15 columns.
- Both the head and tail have the same number of columns indicating that our dataset is consistent.
- Out of the 15 columns, 10 are numeric while 5 are of object type.
- The following columns have missing values:
    - SES - 19
    - MMSE - 2


## Data Preparation

Validity

We chose to drop the following columns:

* ID, Subject ID, MRI ID, Visit - we don't need unique identifiers for our analysis
* Hand -  All subjects chosen for the study were right-handed hence no comparison can be done
* Delay, MR Delay - Only has 20 non-null values out of 436
* Group - We have the CDR column which serves a similar purpose

In [6]:
# Drop irrelevant columns in dataset_1
dataset_1.drop(columns=['ID', 'Hand', 'Delay'], inplace=True)
# Drop irrelevant columns in dataset_2
dataset_2.drop(columns=['Subject ID', 'MRI ID', 'Hand', 'MR Delay', 'Group', 'Visit'], inplace=True)

Consistency

We renamed the EDUC column in dataset_2 to match the Educ column in dataset_1

In [7]:
#Rename the EDUC column
dataset_2 = dataset_2.rename(columns={'EDUC':'Educ'})

Now that we dropped the irrelevant columns and also made the column descriptions consistent, we merged the two datasets

In [8]:
# Merge dataset_1 and dataset_2
merged_df = pd.concat([dataset_1, dataset_2])
merged_df

M/F  Age  Educ  SES  MMSE  CDR  eTIV   nWBV    ASF
0     F   74   2.0  3.0  29.0  0.0  1344  0.743  1.306
1     F   55   4.0  1.0  29.0  0.0  1147  0.810  1.531
2     F   73   4.0  3.0  27.0  0.5  1454  0.708  1.207
3     M   28   NaN  NaN   NaN  NaN  1588  0.803  1.105
4     M   18   NaN  NaN   NaN  NaN  1737  0.848  1.010
..   ..  ...   ...  ...   ...  ...   ...    ...    ...
368   M   82  16.0  1.0  28.0  0.5  1693  0.694  1.037
369   M   86  16.0  1.0  26.0  0.5  1688  0.675  1.040
370   F   61  13.0  2.0  30.0  0.0  1319  0.801  1.331
371   F   63  13.0  2.0  30.0  0.0  1327  0.796  1.323
372   F   65  13.0  2.0  30.0  0.0  1333  0.801  1.317

[809 rows x 9 columns]

We used the DatasetChecker class to look into the various features of the merged_df

In [9]:
# Create an instance of the DatasetChecker class for the third dataset
checker3 = DatasetChecker(merged_df)

# Use a for loop to iterate over the methods in the list
for method in methods:
  # Call the method on the instance, passing no arguments
    result = method(checker3)
    display(result)

First five rows:


M/F  Age  Educ  SES  MMSE  CDR  eTIV   nWBV    ASF
0   F   74   2.0  3.0  29.0  0.0  1344  0.743  1.306
1   F   55   4.0  1.0  29.0  0.0  1147  0.810  1.531
2   F   73   4.0  3.0  27.0  0.5  1454  0.708  1.207
3   M   28   NaN  NaN   NaN  NaN  1588  0.803  1.105
4   M   18   NaN  NaN   NaN  NaN  1737  0.848  1.010

Last five rows:


M/F  Age  Educ  SES  MMSE  CDR  eTIV   nWBV    ASF
368   M   82  16.0  1.0  28.0  0.5  1693  0.694  1.037
369   M   86  16.0  1.0  26.0  0.5  1688  0.675  1.040
370   F   61  13.0  2.0  30.0  0.0  1319  0.801  1.331
371   F   63  13.0  2.0  30.0  0.0  1327  0.796  1.323
372   F   65  13.0  2.0  30.0  0.0  1333  0.801  1.317

Shape of dataset


(809, 9)

Information on dataset:
<class 'pandas.core.frame.DataFrame'>
Int64Index: 809 entries, 0 to 372
Data columns (total 9 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   M/F     809 non-null    object 
 1   Age     809 non-null    int64  
 2   Educ    608 non-null    float64
 3   SES     570 non-null    float64
 4   MMSE    606 non-null    float64
 5   CDR     608 non-null    float64
 6   eTIV    809 non-null    int64  
 7   nWBV    809 non-null    float64
 8   ASF     809 non-null    float64
dtypes: float64(6), int64(2), object(1)
memory usage: 63.2+ KB


None

Summary statistics


Age        Educ        SES        MMSE         CDR         eTIV  \
count  809.000000  608.000000  570.00000  606.000000  608.000000   809.000000   
mean    63.186650   10.184211    2.47193   27.234323    0.288651  1484.782447   
std     23.117511    6.058388    1.12805    3.687980    0.377697   166.911689   
min     18.000000    1.000000    1.00000    4.000000    0.000000  1106.000000   
25%     49.000000    4.000000    2.00000   26.000000    0.000000  1361.000000   
50%     72.000000   12.000000    2.00000   29.000000    0.000000  1475.000000   
75%     80.000000   16.000000    3.00000   30.000000    0.500000  1583.000000   
max     98.000000   23.000000    5.00000   30.000000    2.000000  2004.000000   

             nWBV         ASF  
count  809.000000  809.000000  
mean     0.763037    1.197311  
std      0.059401    0.133031  
min      0.644000    0.876000  
25%      0.715000    1.108000  
50%      0.754000    1.190000  
75%      0.817000    1.290000  
max      0.893000    1.587000

Data types:


M/F      object
Age       int64
Educ    float64
SES     float64
MMSE    float64
CDR     float64
eTIV      int64
nWBV    float64
ASF     float64
dtype: object

Number of missing values


M/F       0
Age       0
Educ    201
SES     239
MMSE    203
CDR     201
eTIV      0
nWBV      0
ASF       0
dtype: int64

We observed missing values in the following columns:
* Educ - 201
* SES - 239
* MMSE - 203
* CDR -201

In [10]:
# Function for checking percentage of missing values
def missing_values(data):
     """
     Identify the missing values and their percentages
     Drop values that have no missing values
     Return only data with missing values
     """
     miss_val = data.isna().sum().sort_values(ascending = False)
     percentage = (data.isna().sum() / len(data)*100).sort_values(ascending = False)
     missing_values = pd.DataFrame({"Missing Values": miss_val, "In Percentage": percentage})
     return missing_values

In [11]:
# check percentage of missing values
missing_values(merged_df)

Missing Values  In Percentage
SES              239      29.542645
MMSE             203      25.092707
Educ             201      24.845488
CDR              201      24.845488
M/F                0       0.000000
Age                0       0.000000
eTIV               0       0.000000
nWBV               0       0.000000
ASF                0       0.000000

Consistency

Since Educ, SES and MMSE have a significant percentage of missing values, we decided to use Linear Regression to replace the values. We will use Measures of Central Tendency to replace missing values in CDR

In [12]:
# Create the testing data fo each column
test_data_educ = merged_df[merged_df['Educ'].isnull()].drop(columns=['M/F','SES','MMSE','CDR'])
test_data_SES = merged_df[merged_df['SES'].isnull()].drop(columns=['M/F','Educ','MMSE','CDR'])
test_data_MMSE = merged_df[merged_df['MMSE'].isnull()].drop(columns=['M/F','Educ','SES','CDR'])
test_data_CDR = merged_df[merged_df['CDR'].isnull()].drop(columns=['M/F','Educ','SES','MMSE'])

# create a dataframe with no missing values
cleaned_df = merged_df.dropna()

# create y_train for each column
y_train_educ = cleaned_df['Educ']
y_train_SES = cleaned_df['SES']
y_train_MMSE = cleaned_df['MMSE']
y_train_CDR = cleaned_df['CDR']

#create x_train for each column
x_train_educ = cleaned_df.drop(['Educ','M/F','SES','MMSE','CDR'], axis=1)
x_train_SES = cleaned_df.drop(['Educ','M/F','SES','MMSE','CDR'], axis=1)
x_train_MMSE = cleaned_df.drop(['Educ','M/F','SES','MMSE','CDR'], axis=1)
x_train_CDR = cleaned_df.drop(['Educ','M/F','SES','MMSE','CDR'], axis=1)

# instantiate models for each column
lr_educ = LinearRegression()
lr_SES = LinearRegression()
lr_MMSE = LinearRegression()
lr_CDR = LinearRegression()

# fit the training data for each model
lr_educ.fit(x_train_educ,y_train_educ)
lr_SES.fit(x_train_SES,y_train_SES)
lr_MMSE.fit(x_train_MMSE,y_train_MMSE)
lr_CDR.fit(x_train_CDR,y_train_CDR)

# create testing data for each column
x_test_educ = test_data_educ.drop('Educ', axis=1)
x_test_SES = test_data_SES.drop('SES', axis=1)
x_test_MMSE = test_data_MMSE.drop('MMSE', axis=1)
x_test_CDR = test_data_CDR.drop('CDR', axis=1)

# predict values for each column
y_pred_educ = lr_educ.predict(x_test_educ)
y_pred_SES = lr_SES.predict(x_test_SES)
y_pred_MMSE = lr_MMSE.predict(x_test_MMSE)
y_pred_CDR = lr_CDR.predict(x_test_CDR)

# replace missing values for each column with the predict values respectively
merged_df.loc[merged_df['Educ'].isnull(), 'Educ'] = y_pred_educ.astype('i')
merged_df.loc[merged_df['SES'].isnull(), 'SES'] = np.round(y_pred_SES, 1)
merged_df.loc[merged_df['MMSE'].isnull(), 'MMSE'] = y_pred_MMSE.astype('i')
merged_df.loc[merged_df['CDR'].isnull(), 'CDR'] = np.round(y_pred_CDR, 1)

# function to round off to the nearest 0.5
def round_to_nearest_half_int(num):
    return round(num * 2) / 2

# round off CDR column to the nearest 0.5
merged_df['CDR'] = merged_df.CDR.apply(lambda x:round_to_nearest_half_int(x))

In [13]:
# Create an instance of the DatasetChecker class for the cleaned merged dataset
checker4 = DatasetChecker(merged_df)

# Use a for loop to iterate over the methods in the list
for method in methods:
  # Call the method on the instance, passing no arguments
    result = method(checker4)
    display(result)

First five rows:


M/F  Age  Educ  SES  MMSE  CDR  eTIV   nWBV    ASF
0   F   74   2.0  3.0  29.0  0.0  1344  0.743  1.306
1   F   55   4.0  1.0  29.0  0.0  1147  0.810  1.531
2   F   73   4.0  3.0  27.0  0.5  1454  0.708  1.207
3   M   28   6.0  2.0  26.0  0.5  1588  0.803  1.105
4   M   18   6.0  1.6  27.0  0.0  1737  0.848  1.010

Last five rows:


M/F  Age  Educ  SES  MMSE  CDR  eTIV   nWBV    ASF
368   M   82  16.0  1.0  28.0  0.5  1693  0.694  1.037
369   M   86  16.0  1.0  26.0  0.5  1688  0.675  1.040
370   F   61  13.0  2.0  30.0  0.0  1319  0.801  1.331
371   F   63  13.0  2.0  30.0  0.0  1327  0.796  1.323
372   F   65  13.0  2.0  30.0  0.0  1333  0.801  1.317

Shape of dataset


(809, 9)

Information on dataset:
<class 'pandas.core.frame.DataFrame'>
Int64Index: 809 entries, 0 to 372
Data columns (total 9 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   M/F     809 non-null    object 
 1   Age     809 non-null    int64  
 2   Educ    809 non-null    float64
 3   SES     809 non-null    float64
 4   MMSE    809 non-null    float64
 5   CDR     809 non-null    float64
 6   eTIV    809 non-null    int64  
 7   nWBV    809 non-null    float64
 8   ASF     809 non-null    float64
dtypes: float64(6), int64(2), object(1)
memory usage: 63.2+ KB


None

Summary statistics


Age        Educ         SES        MMSE         CDR  \
count  809.000000  809.000000  809.000000  809.000000  809.000000   
mean    63.186650    9.056860    2.378863   27.245983    0.242892   
std     23.117511    5.647371    0.971658    3.232349    0.351837   
min     18.000000    1.000000    0.300000    4.000000    0.000000   
25%     49.000000    4.000000    2.000000   27.000000    0.000000   
50%     72.000000    7.000000    2.100000   28.000000    0.000000   
75%     80.000000   14.000000    3.000000   29.000000    0.500000   
max     98.000000   23.000000    5.000000   30.000000    2.000000   

              eTIV        nWBV         ASF  
count   809.000000  809.000000  809.000000  
mean   1484.782447    0.763037    1.197311  
std     166.911689    0.059401    0.133031  
min    1106.000000    0.644000    0.876000  
25%    1361.000000    0.715000    1.108000  
50%    1475.000000    0.754000    1.190000  
75%    1583.000000    0.817000    1.290000  
max    2004.000000    0.893000    1.587000

Data types:


M/F      object
Age       int64
Educ    float64
SES     float64
MMSE    float64
CDR     float64
eTIV      int64
nWBV    float64
ASF     float64
dtype: object

Number of missing values


M/F     0
Age     0
Educ    0
SES     0
MMSE    0
CDR     0
eTIV    0
nWBV    0
ASF     0
dtype: int64

As observed above we have no missing values